### PR TITLE
Remove database and host components.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR follows #575.

- Remove database and host components.
- Update docs/setup files.

## Motivation and Context
Part of signac 2.0 API cleanup.

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [ ] New feature
- [x] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [x] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/master/changelog.txt) and added all related issue and pull request numbers for future reference (if applicable). See example below.

### CODE REVIEW

**Comment by mikemhenry** [comment]
tasty
+13 −1,067

**Comment by bdice** [comment]
@vyasr @tommy-waltmann I assigned you as reviewers because Brandon (the suggested committer) is out this week and you two were some of the last developers to touch these files. This PR is a dependency of #580, which I will re-work once this PR is merged.

**Review by tommy-waltmann — APPROVED** [review]

**Pull request merged**
